service/mapper/transitions.go

```diff
@@ -269,9 +269,116 @@ func (t *Transitions) ResumeIndexing(s *State) error {
 	// then proceed with the first non-indexed block and index values
 	s.height = last + 1
 
-	// At this point, we should be able to start indexing the chain data for
+	// At this point, we should be able to start indexing the register data for
 	// the next height.
-	s.status = StatusIndex
+	s.status = StatusUpdate
+	return nil
+}
+
+// UpdateTree gets all trie updates and stores it in the state
+func (t *Transitions) UpdateTree(s *State) error {
+	if s.status != StatusUpdate {
+		return fmt.Errorf("invalid status for updating tree (%s)", s.status)
+	}
+	log := t.log.With().Uint64("height", s.height).Logger()
+	// grab updates and move on to next state
+	updates, err := t.updates.AllUpdates()
+	if errors.Is(err, archive.ErrUnavailable) {
+		time.Sleep(t.cfg.WaitInterval)
+		log.Debug().Msg("waiting for next trie update")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("unable to retrieve trie updates for block height %x", s.height)
+	}
+	s.updates = updates
+	log.Info().Int("updates", len(s.updates)).Msg("collected trie updates (registers) to be mapped")
+
+	// Now that we have collected TrieUpdates for the current block,
+	// we can create a path -> payload mapping that we will use to index to the local storage
+	s.status = StatusCollect
+	return nil
+}
+
+// CollectRegisters reads the payloads for the next block to be indexed from the state's forest, unless payload
+// indexing is disabled.
+func (t *Transitions) CollectRegisters(s *State) error {
+	if s.status != StatusCollect {
+		return fmt.Errorf("invalid status for collecting registers (%s)", s.status)
+	}
+
+	// If indexing payloads is disabled, we can bypass collection and indexing
+	// of payloads and just go straight to indexing chain data
+	if t.cfg.SkipRegisters {
+		s.status = StatusIndex
```

Comment:
do we need to call `LatestRegisterHeight` here?

Reply:
Thought about this, and I guess if there are registers for blocks 0-100 and no registers for block 101, we should still update latest as 101

Reply:
I guess it depends on what `LatestRegisterHeight` is tracking. If it's tracking the last height with a register change, the we shouldn't. But if it's tracking the latest height where register updates were indexed, then it should.